wled00/FX.cpp

```diff
@@ -5582,16 +5587,14 @@ static const char _data_FX_MODE_2DPLASMABALL[] PROGMEM = "Plasma Ball@Speed,,Fad
 ////////////////////////////////
 //  2D Polar Lights           //
 ////////////////////////////////
-//static float fmap(const float x, const float in_min, const float in_max, const float out_min, const float out_max) {
-//  return (out_max - out_min) * (x - in_min) / (in_max - in_min) + out_min;
-//}
-uint16_t mode_2DPolarLights(void) {        // By: Kostyantyn Matviyevskyy  https://editor.soulmatelights.com/gallery/762-polar-lights , Modified by: Andrew Tuline
+
+uint16_t mode_2DPolarLights(void) {        // By: Kostyantyn Matviyevskyy  https://editor.soulmatelights.com/gallery/762-polar-lights , Modified by: Andrew Tuline & @dedehai (palette support)
   if (!strip.isMatrix || !SEGMENT.is2D()) return mode_static(); // not a 2D set-up
 
   const int cols = SEGMENT.virtualWidth();
   const int rows = SEGMENT.virtualHeight();
 
-  CRGBPalette16 auroraPalette  = {0x000000, 0x003300, 0x006600, 0x009900, 0x00cc00, 0x00ff00, 0x33ff00, 0x66ff00, 0x99ff00, 0xccff00, 0xffff00, 0xffcc00, 0xff9900, 0xff6600, 0xff3300, 0xff0000};
+ // CRGBPalette16 auroraPalette  = {0x000000, 0x003300, 0x006600, 0x009900, 0x00cc00, 0x00ff00, 0x33ff00, 0x66ff00, 0x99ff00, 0xccff00, 0xffff00, 0xffcc00, 0xff9900, 0xff6600, 0xff3300, 0xff0000};
```

Comment:
This will cause *different looking* effect. Please consider backwards compatibility.

You should keep this if the `SEGMENT.palette == 0`, a default value for palette.

Reply:
I fixed this FX to be useable with any palette, there is no longer the requirement of 'black edges' as it was before. But I agree that adding the palette to the list of palettes makes more sense. I will see what I can come up with.

Reply:
> I disagree, if this palette is so revolutionary, it should be available to every effect, not just this one.

Do not forget: Most effects were ported from FastLED effects. Not everyone wanted (or knew how) to make them WLED native. Also, there wasn't a generic way to specify default values for an effect (metadata). Things have changed since.

Reply:
I wasn't baling anyone for anything, just saying that we should either remove or move it, but not keep it here

Reply:
I moved it to a palette in latest commit